### PR TITLE
Add mention notifications to activity center & empty state

### LINKED ISSUE #12064: Show mentions in activity center
As a user
I want to see messages that mentions me in the activity center
So that is easier to find the relevant posts

### Descriptions
Mentions should be shown in the activity center as per designs

https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1405%3A345

### LINKED ISSUE #12237: Empty state in activity center
As a user
I want to see an empty state in activity center
So that I explicitly know if I don't have any activity to interact with

### Descriptions
Empty state be shown in the activity center as per designs

https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1405%3A345

### LINKED ISSUE #12238: Group and sort notifications by date in activity center
As a user
I want to see notifications grouped and sorted by date in activity center
So that I can better see what activity I have for each date

### Descriptions
Notification groups should be shown in the activity center as per designs

https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1405%3A345

### LINKED ISSUE #12250: Unexpected notification in Activity Center when mention of my account received in Group Chat
# Bug Report

## Problem 
When user receives mention of own account into already joined Group Chat - there is notification displayed on Activity Center. Nothing in Activity Center view when opening it. 
All fine in 1-1 / Public chats.
All fine if to send any other (text/image/sticker) message

#### Expected behavior
No notifications in Activity Center when mention of my account received in any Group Chat
#### Actual behavior
Notification on Activity Center icon when mention of my account received in any Group Chat

https://user-images.githubusercontent.com/8749671/123292964-37637e80-d51c-11eb-971a-8637efc68a1b.mov

### Reproduction

1) Create two new accounts: UserA and UserB
2) UserA adds UserB in contacts
3) UserA creates Group Chat with UserB
4) UserB joins into GroupChat
5) UserA sends mentioned UserB in message
6) UserB verifies Activity Center notification icon on Chats view

### Additional Information

- Status version: develop 1.14.0 (20210624055928) 
- Operating System: Both Android and iOS

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #12064
fixes #12237
fixes #12238
fixes #12250

### Summary

This PR adds mention notification to activity center. Based on designs, there are a couple of things that needs to be refactored to fully comply with the whole activity center feature expected UI/UX, and will be handled on a series of different PRs. Some things that still need to be worked are:
- The Select feature is still always visible under the top bar
- Notifications are always immediately marked as read when navigating to the activity center, and that may brake highlighting a new notification. The only way to view a notification in a highlighted background is to receiving a notification while already being in the activity center screen
- Top bar still needs more work
- Ellipsis on both sides are a bit tricky to implement, so didn't implement it on this PR 

However, I took advantage and added a bit more to the scope for the sake of UI consistency:
- Unify UI styling for different type of notifications
- Group and sort notifications by date
- Added an empty state to the activity center screen

### Testing notes

- Mentions only work for groups and communities for the time being

#### Platforms

- Android
- iOS

##### Functional

- activity center

### Steps to test

- Open Status
- Open a group or community chat
- Send a message mentioning some user

status: ready

### CODE REVIEW

**Comment by Serhy** [comment]
@briansztamfater could you rebase against latest develop please?

**Comment by qoqobolo** [comment]
@briansztamfater thanks for PR!

### ISSUE 1: Unread indicator for new mentions on the Activity center bell doesn't disappear if open chat with received mention from the Home screen

**_Steps:_**
**UserA:**
- Create a group chat with UserB

**UserB:**
- Join the group
- Navigate to the Home screen

**UserA:**
- Mention UserB

**UserB:**
- Open the chat with mention
- Return to the Home screen
- Check indicator on the bell

**Expected result:** the new mention indicator disappears after opening a chat containing it, no matter what screen it was opened from
**Actual result:** the new mention indicator disappears only after opening the activity center


https://user-images.githubusercontent.com/67952253/123262896-aaf39480-d4f8-11eb-927d-b4ddeee07db4.mp4



### ISSUE 2: Notifications for mentions in the activity center are duplicated if they are received from different chats (it can be group and community, or two communities) while the User is in the activity center screen, and if the activity center already contains at least two notifications from one type of chat

**_Steps:_**

**UserA:**
1. Create a group chat and community

**UserB:**
2. Join both
3. Open Activity center

**UserA:**
4. Send two mentions of UserB to the community channel
5. Send one mention of UserB to the group chat

**UserB:**
6. Check the number of notifications

**OS:** iOS, Android

_On the video below only one mention in the community was received. The two top notifications from the group chat are duplicates._

https://user-images.githubusercontent.com/67952253/123252401-85608e00-d4ec-11eb-94f4-d5e2bdfe79f3.mp4

While attempting to open duplicated notification get error
Screenshot 2021-06-24 at 17 25 20

[geth.log](https://github.com/status-im/status-react/files/6710428/geth.log)
[Status.log](https://github.com/status-im/status-react/files/6710430/Status.log)

It seems like an edge case. 
But not sure if it can affect the notifications in the future if the issue with the highlighted notifications is fixed and the User can receive them not only while on the screen of the activity center?
WDYT?


### ISSUE 3: Double `@` in the mention notification received from an ENS user

Screenshot 2021-06-24 at 15 27 35

**Comment by briansztamfater** [comment]
@qoqobolo Thanks for pointing out these issues, all of them make sense to fix and working on them right now.

**Comment by briansztamfater** [comment]
@qoqobolo issues should be fixed. Let me know how it goes :)

**Comment by qoqobolo** [comment]
@briansztamfater thank you! Unfortunately, **issue 1** and **issue 3** are still reproducible :(

**Issue 1** is fixed only for group chats. Also, this issue is relevant for cases when a user receives a mention while in the chat view (both group chat and community).

**Issue 3**
Still reproducible.
Probably, `@` is also superfluous if the message was received from a user with a 3-random name.
Screenshot 2021-06-25 at 12 53 47

**Comment by briansztamfater** [comment]
@qoqobolo you are right, made new changes and hopefully they should be fixed now

**Comment by qoqobolo** [comment]
@briansztamfater thank you, looks great now!

Tested:
- Group and sort notifications by date in activity center
- Empty state in activity center
- Mentions in activity center (for communities and group chats)
- Highlighted notifications
- Mentions from different types of chats received at the same time
- ENS
- On mobile network; offline > online
- Changes in a community or group chat (e.g. name of community/channel/group) are updated in labels in activity center
- Notifications from contact/not contact/blocked/muted user
- No notifications after leaving the community

PR is ready to be merged.

**Comment by qoqobolo** [comment]
@briansztamfater thanks, looks great now!

Could you add accessibility-ids for these elements, please?

Screenshot 2021-06-28 at 12 30 17

After that, PR will be ready to merge.

Tested:

- Group and sort notifications by date in activity center
- Empty state in activity center
- Mentions in activity center (for communities and group chats)
- Highlighted notifications
- Mentions from different types of chats received at the same time
- ENS
- On mobile network; offline > online
- Changes in a community or group chat (e.g. name of community/channel/group) are updated in labels in activity center
- Notifications from contact/not contact/blocked/muted user
- No notifications after leaving the community

**Comment by briansztamfater** [comment]
@qoqobolo Thanks, added accessibility ids and rebased, let me know if you need anything else :)

I need a code review approve in order to merge so maybe I'll request a quick re-test in case we need to change anything.

**Review by shivekkhurana — APPROVED** [review]
Looks good.

File: src/status_im/ui/screens/notifications_center/views/notification.cljs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          (fn [{:keys [_ length] :as acc-text} new-text-chunk]
+            (if (>= length max-length)
+              (reduced acc-text)
+              (add-parsed-to-message acc-text new-text-chunk)))
+          {:components (message-wrapper number-of-lines)
+           :length     0}
+          parsed-text)]
+     (:components result))))
+
+(defn message-content-text [{:keys [content content-type community-id]}]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/notifications_center/views/notification.cljs:72]: just wondering if this duplicated from home screen? could we reuse functions from home ?

Comment (briansztamfater) [reply src/status_im/ui/screens/notifications_center/views/notification.cljs:72]: The styles and some minor logic is different, i.e. `mention-element` that always show an `@`, and if there's no message on the notification we show the text in gray

**Pull request merged**